File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        num,sum=0,0
        for i in str(n):
            if(int(i)!=0):
                num*=10
                num+=(int(i))
                sum+=int(i)
        num*=sum
        return num

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        String s=Integer.toString(n);
        long num=0;
        long sum=0;
        for(int i=0; i<s.length(); i++){
            int ch=s.charAt(i)-'0';
            if(ch>0){
                sum+=ch;
                num*=10;
                num+=ch;
            }
        }
        num*=sum;
        return num;
    }
}